pkg/asset/manifests/openstack/cloudproviderconfig.go

```diff
@@ -54,27 +50,41 @@ func CloudProviderConfigSecret(cloud *clientconfig.Cloud) ([]byte, error) {
 		domainName = cloud.AuthInfo.UserDomainName
 	}
 
-	config := &config{
```

Comment:
I think we shouldn't have moved away from typed config at least. A helper function to go from typed config to ini file would have been more suitable imo.